autoscale/src/main/java/com/sequenceiq/periscope/service/security/TlsSecurityService.java

```diff
@@ -34,11 +36,19 @@ public TlsConfiguration getConfiguration(Cluster cluster) {
         if (securityConfig != null) {
             return new TlsConfiguration(securityConfig.getClientKeyDecoded(), securityConfig.getClientCertDecoded(), securityConfig.getServerCertDecoded());
         }
-        securityConfig = securityConfigRepository.findByClusterId(cluster.getId());
+        securityConfig = getSecurityConfigSilently(cluster);
```

Comment:
please change swap the prepareSecurityConfig and getConfiguration methods with each other to avoid upward referencing/calling